add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt

```diff
@@ -0,0 +1,41 @@
+package dev.flutter.multipleflutters
+
+import java.lang.ref.WeakReference
+
+/**
+ * Interface for getting notifications when the DataModel is updated.
+ */
+interface DataModelObserver {
+    fun onCountUpdate(newCount : Int)
+}
+
+/**
+ * A singleton/observable data model for the data shared between Flutter and the host platform.
+ *
+ * This is the definitive source of truth for all data.
+ */
+class DataModel {
+    companion object {
+        val instance = DataModel()
+    }
+
+    private val observers =  mutableListOf<WeakReference<DataModelObserver>>()
```

Comment:
Kotlin ftw!!

Reply:
Oh man, so much easier in this case.  Swift's protocols are weird.